nvi-commons/src/main/java/no/sikt/nva/nvi/common/db/NviPeriodRepository.java

```diff
@@ -29,4 +34,13 @@ public Optional<DbNviPeriod> findByPublishingYear(String publishingYear) {
         var fetched = this.nviPeriodTable.getItem(queryObj);
         return Optional.ofNullable(fetched).map(NviPeriodDao::nviPeriod);
     }
+
+    public List<DbNviPeriod> getPeriods() {
+        return this.nviPeriodTable.scan().stream()
+                   .map(Page::items)
+                   .flatMap(Collection::stream)
+                   .map(NviPeriodDao::nviPeriod)
+                   .filter(Objects::nonNull)
+                   .toList();
+    }
```

Comment:
Not sure if this scan scans all items in table, or only those of type PERIOD.

Reply:
From what I can tell, it scans all items.

Try something like:

```
        var expression = Expression.builder()
                                      .expression("#a = :b")
                                      .putExpressionName("#a", "type")
                                      .putExpressionValue(":b", AttributeValue.fromS("PERIOD"))
                                      .build();
        var scanEnhancedRequest = ScanEnhancedRequest.builder()
                    .filterExpression(expression)
                    .build();

        return this.nviPeriodTable.scan(scanEnhancedRequest).stream()
                   .map(Page::items)
                   .flatMap(Collection::stream)
                   .map(NviPeriodDao::nviPeriod)
                   .filter(Objects::nonNull)
                   .toList();
```